OCR/JSTree, UV Integration

haydenmccormick opened the pull request:
Addition of OCR and JSTree features, as well as UV integration and refactoring.

marcverhagen reviewed (commented):
Looks good to me in general. Two small comments:

- In the Dockerfile it is probably better to move the two apt-get commands above copying the requirments. This is assuming that changes to Python imports are more likely than changes to the apt-get installations.
- In general we have a convention to name the main script app.py, here it is routes.py.

haydenmccormick commented:
No problem: addressed with 9ef0630a6bdf1842ff97770f73fcf9c3788a25bf
Taking a step back when testing again, I also noticed that the tab name "Interactive_MMIF" is probably unintuitive and a bit ugly, so I changed it to just "Tree."
